modules/PatternUtils.js

```diff
@@ -21,13 +21,13 @@ function _compilePattern(pattern) {
     }
 
     if (match[1]) {
-      regexpSource += '([^/?#]+)'
+      regexpSource += '([^/]+)'
```

Comment:
This is all legacy crap. History deals with all of this now; you're just not going to get unescaped `?`s and `#`s in the path any more.